### PR TITLE
Adds a basic JsonStreamQueryTransformer

### PR DESCRIPTION
**Pull request by Tpt**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,10 +74,10 @@
     ],
     "coverageThreshold": {
       "global": {
-        "branches": 94,
+        "branches": 95,
```
Comment (rubensworks) [inline package.json:77]: Can we make this all 100? :)

**Review by rubensworks — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Tpt) [reply package.json:77]: Yes, would be great! jsonparse code coverage was not great I already improved it a bit with `[JSONTestSuite](https://github.com/nst/JSONTestSuite/tree/d64aefb55228d9584d3e5b2433f720ea8fd00c82)` but there is indeed still work to do. But it's unrelated to this PR imho (the introduced code as a 100% code coverage).

File: lib/JsonStreamQueryTransformer.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+export interface IQueryResult {
+  query: string;
+  value?: JsonValue;
+}
+
+/**
+ * Emits all JSON blobs matching one of the given query.
+ * A query is a path composed of object keys or `null` for array keys.
+ */
+export class JsonStreamQueryTransformer extends Transform {
```
[the remark this answers is not in the document]
Comment (Tpt) [reply lib/JsonStreamQueryTransformer.ts:23]: Done with `JsonStreamPathTransformer`. What is implemented is a subset of JSONPath so it should be possible to add JSONPath without breaking the API.

**Review by rubensworks — APPROVED** [review]
Looks good to me!

**Pull request merged**